### PR TITLE
Atmire 2344 association anywhere rewrite

### PR DESCRIPTION
**Pull request by mdiggory**
This is a rewrite of the association anywhere utility code to better handle xml parsing and correct missing features in specification. It completely replaces #722

### CODE REVIEW

**Comment by mdiggory** [comment]
Failures in travis are related to the changes to the workflow we will need to adjust workflow tests further..

Tests in error: 
  testApproveFromBlackoutWithPastDate(org.dspace.workflow.AutoApproveBlackoutProcessorTest)
  testApproveFromBlackoutFailsWithFutureDate(org.dspace.workflow.AutoApproveBlackoutProcessorTest)
  testApproveFromBlackoutFailsWithNoDate(org.dspace.workflow.AutoApproveBlackoutProcessorTest)

I will address these next.

File: dspace-xmlui/dspace-xmlui-api/src/main/java/org/dspace/app/xmlui/aspect/authority/concept/ManageConceptMain.java [hunk 1]
```diff
@@ -140,7 +143,14 @@ public void addBody(Body body) throws WingException, SQLException
...
         int page          = parameters.getParameterAsInteger("page",0);
         int highlightID   = parameters.getParameterAsInteger("highlightID",-1);
-        String query      = decodeFromURL(parameters.getParameter("query",null));
+        String query_1      = parameters.getParameter("query", null);
+        String query = "";
+        try{
+            query= URLDecoder.decode(query_1, Constants.DEFAULT_ENCODING);
+        }catch (UnsupportedEncodingException e)
+        {
+            log.error("decode error:"+e);
```
Comment (ryscher) [inline dspace-xmlui/dspace-xmlui-api/src/main/java/org/dspace/app/xmlui/aspect/authority/concept/ManageConceptMain.java]: It's better to write this with the exception passed as an argument, so the logs contain the stack trace and not just a simple message.

`log.error("decode error:", e);`

File: dspace/modules/api/src/main/java/org/datadryad/anywhere/AssociationAnywhere.java [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+        Document customerInfo = loadCustomerInfo(customerId);
+        String result = printDocument(customerInfo);
+
+        String customerName = getStringValue(customerInfo, "//displayName");
+        String classSubclassDescr = getStringValue(customerInfo, "//classSubclassDescr");
+        String statusCode = getStringValue(customerInfo, "//statusCode");
+
+        //set integrated, make sure integrated is true
+        changeConceptMetadataValue(concept,"customerId",customerId);
+        changeConceptMetadataValue(concept,"integrated","true");
```
Comment (ryscher) [inline dspace/modules/api/src/main/java/org/datadryad/anywhere/AssociationAnywhere.java:280]: Not sure why this is always set to true -- there are journals who appear in AA, but do not have submission integration with Dryad.

Comment (mdiggory) [inline dspace/modules/api/src/main/java/org/datadryad/anywhere/AssociationAnywhere.java:280]: Do we have a field in association anywhere that can be leveraged for "integrated" or would you like to set it manually via Concept metadata?

**Comment by mdiggory** [comment]
superseded by #901

**Pull request closed without merging**